File: examples/tuple_key_dict.py
from typing import Dict, Tuple

def test_tuple_keys() -> Dict[Tuple[int, int], str]:
    """Test dictionary with tuple keys"""
    d: Dict[Tuple[int, int], str] = {}
    
    # Assign values using tuple keys
    d[(0, 0)] = "origin"
    d[(1, 0)] = "right"
    d[(0, 1)] = "up"
    d[(1, 1)] = "diagonal"
    
    # Build tuple dynamically
    x = 2
    y = 3
    d[(x, y)] = "dynamic"
    
    return d

File: examples/tuple_key_dict.rs
use std::collections::HashMap;
#[doc = "Test dictionary with tuple keys"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_tuple_keys() -> HashMap<(i32, i32), String> {
    let mut d: std::collections::HashMap<(i32, i32), String> = {
        let map = HashMap::new();
        map
    };
    d.insert((0, 0), "origin".to_string());
    d.insert((1, 0), "right".to_string());
    d.insert((0, 1), "up".to_string());
    d.insert((1, 1), "diagonal".to_string());
    let x = 2;
    let y = 3;
    d.insert((x, y), "dynamic".to_string());
    d
}
